mods/ra/maps/sarin-gas-2-down-under/downunder.lua

```diff
@@ -0,0 +1,457 @@
+			Spy.Move(SpyMove.Location)
+			Trigger.AfterDelay(DateTime.Seconds(10), function()
+				DogCrewCamera.Destroy()
+			end)
+			Utils.Do(DogCrew, function(actor)
+				if actor.IsDead then
+					return
+				else
+					actor.AttackMove(DoggyMove.Location)
+				end
+			end)
+		end
+	end)
+end
+
+PrisonEscape = function()
+	Trigger.OnEnteredFootprint(PrisonAlarm, function(unit, id)
+		Trigger.RemoveFootprintTrigger(id)
+		Media.DisplayMessage("Warning, prisoners are attempting to escape!", "Intercom")
+		Media.PlaySoundNotification(greece, "AlertBuzzer")
+		Utils.Do(GuardDogs, function(actor)
+			if actor.IsDead then
+				return
+			else
+				actor.Hunt()
+			end
+		end)
+	end)
+end
+
+ScientistExecution = function()
+	Media.PlaySoundNotification(greece, "AlertBleep")
+	Media.DisplayMessage("The base is compromised. We have to hurry the execution!", "Soviet Officer")
+	Utils.Do(DemoTeam, function(actor)
+		actor.AttackMove(DemoDrive2.Location)
+	end)
+
+	Trigger.OnAllKilled(FiringSquad, function()
+		if not ScientistMan.IsDead then
+			ScientistRescued()
+		end
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(7), function()
+		if not Officer2.IsDead then
+			Media.DisplayMessage("Prepare to Fire!", "Soviet Officer")
+		end
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(15), function()
+		if not Officer2.IsDead then
+			Media.DisplayMessage("Fire!", "Soviet Officer")
+		end
+
+		Utils.Do(FiringSquad, function(actor)
+			if actor.IsDead then
+				return
+			else
+				actor.Attack(ScientistMan, true, true)
+			end
+		end)
+	end)
+end
+
+ScientistRescued = function()
+	Media.DisplayMessage("Thanks for the rescue!", "Scientist")
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		if not ScientistMan.IsDead and not DemoTruck.IsDead then
+			Media.DisplayMessage("The Soviets have an unstable nuclear device stored here. \n I need to move it out of the facility!", "Scientist")
+			DemoTruck.GrantCondition("mission")
+			ScientistMan.EnterTransport(DemoTruck)
+		end
+	end)
+
+	Trigger.OnRemovedFromWorld(ScientistMan, DemoTruckExit)
+end
+
+DemoTruckExit = function()
+	if ScientistMan.IsDead then
+		return
+	end
+
+	Media.DisplayMessage("I hope the exit is clear!", "Scientist")
+	Utils.Do(DemoTruckPath, function(waypoint)
+		DemoTruck.Move(waypoint.Location)
+	end)
+
+	Trigger.OnEnteredFootprint({ DemoDrive4.Location }, function(actor, demotruckescape)
+		if actor.Type == "dtrk" then
+			Trigger.RemoveFootprintTrigger(demotruckescape)
+			actor.Destroy()
+		end
+	end)
+end
+
+AcceptableLosses = 0
+Tick = function()
+	if greece.HasNoRequiredUnits() then
+		greece.MarkFailedObjective(ExitBase)
+	end
+
+	if Map.LobbyOption("difficulty") == "hard" and greece.UnitsLost > AcceptableLosses then
+		greece.MarkFailedObjective(NoCasualties)
+	end
+end
+
+WorldLoaded = function()
+	greece = Player.GetPlayer("Greece")
+	england = Player.GetPlayer("England")
+	turkey = Player.GetPlayer("Turkey")
+	ussr = Player.GetPlayer("USSR")
+	badguy = Player.GetPlayer("BadGuy")
```

Comment:
This doesn't seem to be used. (In general I would try to remove as many [unnecessary] players from map.yaml as well, e.g. I guess all of badguy's units can be owned by ussr and the badguy player removed.)

Reply:
The reason for having the badguy player is so the Turkish teslas will only target the badguy mammoth and not a spy with a USSR disguise. I changed all the badguy actors sans said mammoth to USSR.